src/Init/Data/Int/DivModLemmas.lean

```diff
@@ -1194,6 +1194,16 @@ theorem bmod_sub_bmod_congr : Int.bmod (Int.bmod x n - y) n = Int.bmod (x - y) n
     rw [Int.sub_eq_add_neg, Int.sub_eq_add_neg, Int.add_right_comm, ←Int.sub_eq_add_neg, ← Int.sub_eq_add_neg]
     simp [emod_sub_bmod_congr]
 
+theorem add_bmod_eq_add_bmod_right (i : Int)
+    (H : bmod x n = bmod y n) : bmod (x + i) n = bmod (y + i) n := by
+  rw [← bmod_add_bmod_congr, ← @bmod_add_bmod_congr y, H]
+
+theorem bmod_add_cancel_right (i : Int) : bmod (x + i) n = bmod (y + i) n ↔ bmod x n = bmod y n :=
+  ⟨fun H => by
+    have := add_bmod_eq_add_bmod_right (-i) H
+    rwa [Int.add_neg_cancel_right, Int.add_neg_cancel_right] at this,
+  fun H => by rw [← bmod_add_bmod_congr, H, bmod_add_bmod_congr]⟩
```

Comment:
I haven't checked, but are there analogues of these for the other `*mod` variants? I would like to try to keep the discrepancy between the versions monotonically decreasing if possible. :-)

Reply:
Yes, as I mention in the commit message, these two are direct copies of their emod equivalents in both name and proof. bmod_neg_bmod is new as there isn't

Reply:
Ah, sorry I wasn't clear. The request was if you'd be willing to also provide the `*mod` variants of your new `bmod_neg_bmod` lemma. (It's fine if not, but then it needs to be added to a TODO list.)

Reply:
Ah, understood. It does not hold for `tmod`,
```
#eval (-1).tmod 2 == (-3).tmod 2
#eval (-1 + 2).tmod 2 != (-3 + 2).tmod 2
```

For `fmod` it holds for small values so I expect it holds generally. I took a look at doing this; however, the theory for fmod is not well developed, so proving this requires building a lot of groundwork to prove basic results about addition under fmod that I don't think I'd like to take on right now.

Reply:
No problem, thanks. I'm planning on making this more uniform later in the quarter, so I can take it from here.